serpentTools/parsers/results.py

```diff
@@ -1,15 +1,234 @@
+        refVersion = {'meta': 'VERSION ',
+                    'rslt': 'MIN_MACROXS',
+                    'univ': 'GC_UNIVERSE_NAME',
+                    'days': 'BURN_DAYS',
+                    'burn': 'BURNUP',
+                    'inxs': 'INF_',
+                    'b1xs': 'B1_'}
+        MapStrVersions = {}
+        for iVer in range(10, 31):
+            # create a new dictionary for each Serpent version
+            MapStrVersions['2.1.{}'.format(iVer)] = dict(refVersion)
+        # update any change manually
+        MapStrVersions['2.1.30']['rslt'] = 'MEAN_SRC_WGT'
+        if rc['serpentVersion'] in MapStrVersions.keys():
+            self._keysVersion = MapStrVersions[rc['serpentVersion']]
+        else:
+            raise SerpentToolsException("Version {} is not supported by the "
+                                        "ResultsReader".format(rc['serpentVersion']))
+
+    def _preRead(self):
+        """do a quick scan to count the number of universes."""
+        with open(self.filePath) as fid:
+            if fid is None:
+                raise IOError("Attempting to read on a closed file.\n"
+                              "Parser: {}\nFile: {}".format(self, self.filePath))
+            for tline in fid:
+                if self._keysVersion['univ'] in tline:
+                    self._getVarValues(tline)  # obtain universe name
+                    self._univlist.append(self._varVals)
+                    nUniv = self._getUniqueList()
+                    if len(self._univlist) > nUniv:
+                        self._counter['univ'] = nUniv
+                        self._univlist = list()
+                        return
+            if not self._univlist:
+                raise SerpentToolsException("No universes are found in the "
+                                            "file {}".format(self.filePath))
+
+    def _precheck(self):
+        """do a quick scan to ensure this looks like a results file."""
+        with open(self.filePath) as fid:
+            for line in fid:
+                tline = line.split()
+                if not tline:
+                    continue
+                elif 'ABS_KEFF' in tline:
+                    return
+            raise SerpentToolsException("Criticality values were not "
+                                        "found in {}".format(self.filePath))
+
+    def _postcheck(self):
+        """ensure the parser grabbed expected materials."""
+        if not self.resdata:
+            raise SerpentToolsException("No time-dependent results obtained "
+                                        "from {}".format(self.filePath))
+        if divmod(self._counter['meta'],self._counter['univ'])[0] != self._counter['rslt']:
+            debug('  found {} universes, {} time-points, and {} overall result points'.
+                  format(self._counter['univ'], self._counter['rslt'], self._counter['meta']))
+            raise SerpentToolsException("The file {} is not complete ".format(self.filePath))
```

Comment:
Would this be raised if not universes are present in the file, but the file is in fact complete?

Reply:
This was not changed, however the postcheck function was slightly modified:
        if not self.resdata and not self.metadata and not self.univdata:
            raise SerpentToolsException("No results were collected "
                                        "from {}".format(self.filePath))

Reply:
The debug info is now included in the exception. 

            raise SerpentToolsException(
                "The file {} is not complete. The reader found {} universes, "
                "{} time-points, and {} overall result points ".format(self.filePath,
                self._counter['univ'], self._counter['rslt'], self._counter['meta']))